common/scala/src/main/scala/whisk/common/TransactionId.scala

```diff
@@ -27,33 +27,95 @@ import spray.json.JsArray
 import spray.json.JsNumber
 import spray.json.JsValue
 import spray.json.RootJsonFormat
+import java.time.Duration
+import akka.event.Logging.LogLevel
+import akka.event.Logging.{ InfoLevel, WarningLevel }
 
 /**
  * A transaction id for tracking operations in the system that are specific to a request.
- * An instance of TransactionId is implicitly received by all logging methods. This is a
- * value type so that the type system ensures it is never null. The actual metadata is stored
- * indirectly in the referenced meta object.
+ * An instance of TransactionId is implicitly received by all logging methods. The actual
+ * metadata is stored indirectly in the referenced meta object.
  */
-case class TransactionId private (meta: TransactionMetadata) extends AnyVal {
+case class TransactionId private (meta: TransactionMetadata) extends Logging {
```

Comment:
Experimented with this suggestion here: https://github.com/openwhisk/openwhisk/commit/2714f8bf66ac4242a6e53d3c58065ef768634fcb.